basetest.pm

```diff
@@ -715,6 +726,47 @@ sub parse_serial_output_qemu {
     return;
 }
 
+sub search_for_expected_autoinst_failures {
+    my ($self) = @_;
+    # autoinst failures defined in distri (test can override them)
+    my $failures = $self->{autoinst_failures};
+
+    my $die = 0;
+    my %regexp_matched;
+    # loop line by line
+    open(my $AUTOINSTLOG, "<", "autoinst-log.txt");
+    seek($AUTOINSTLOG, $autoinst_log_pos, 0);
+    while (my $line = <$AUTOINSTLOG>) {
+        chomp $line;
+        for my $regexp_table (@{$failures}) {
+            my $regexp  = $regexp_table->{pattern};
+            my $message = $regexp_table->{message};
+            my $type    = $regexp_table->{type};
+
+            # Input parameters validation
+            die "Wrong type defined for serial failure. Only 'soft' or 'hard' allowed. Got: $type" if $type !~ /^soft|hard|fatal$/;
+            die "Message not defined for serial failure for the pattern: '$regexp', type: $type" if !defined $message;
+
+            # If you want to match a simple string please be sure that you create it with quotemeta
+            if (!exists $regexp_matched{$regexp} and $line =~ /$regexp/) {
+                $regexp_matched{$regexp} = 1;
+                my $fail_type = 'softfail';
+                if ($type =~ 'hard|fatal') {
+                    $die                   = 1;
+                    $fail_type             = 'fail';
+                    $self->{fatal_failure} = $type eq 'fatal';
+                }
+                $self->record_resultfile($message, $message . " - autoinst-log error: $line", result => $fail_type);
+                $self->{result} = $fail_type;
+            }
+        }
+    }
+    $autoinst_log_pos = tell $AUTOINSTLOG;
```

Comment:
If the previous open/seek or reads fails this will assign `-1` to that global variable. When calling `search_for_expected_autoinst_failures` again this would be used for the initial seek. This behavior is likely not the easiest to debug. Since it is not nice to suppress errors silently anyways I suggest to handle possible errors.

Reply:
Checking error for seek and open.

Reply:
But not for the reading the lines. If a read fails the while loop ends and I think `tell` would also return `-1` then.

Reply:
fixed that.